python/cudf/cudf/core/dataframe.py

```diff
@@ -821,6 +822,34 @@ def _from_data(
             out.columns = columns
         return out
 
+    def _align_to_index(
+        self, index, how="outer", sort=True, allow_non_unique=False
+    ):
+        # TODO: support other kwargs
+        index = as_index(index)
+        if self.index.equals(index):
+            return self
+
+        lhs = self.copy(deep=False)
+        rhs = cudf.DataFrame(index=as_index(index))
+
+        rng_col_id = str(uuid4())
+        if how == "left":
+            lhs[rng_col_id] = column.arange(len(lhs))
+        else:
+            rhs[rng_col_id] = column.arange(len(rhs))
```

Comment:
It seems like this gets allocated and then unused in the `how=outer` case. If that's the case this might be worth addressing since it is the default case.